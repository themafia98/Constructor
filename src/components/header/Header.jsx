import React from 'react';

import PropTypes from 'prop-types';
import './header.scss';

import {middlewareLogOutUser} from '../../redux/middleware/loadUserMiddleware';
import {connect} from 'react-redux';
import {Redirect, withRouter} from 'react-router-dom';

import eventEmitter from '../../EventEmitter.js';
import Icon from '../Icon/icon';


const iconPath = require('../../config.json').CabinetIcon;

class Header extends React.Component {

    static propTypes = {
        title: PropTypes.string.isRequired,
    };

    state = {
        title: this.props.title,
        redirectAbout: {
            path:  '/Cabinet/About',
            redirectA: false,
        },
        redirectCabinet: {
            path:  '/Cabinet',
            redirectC: false,
        }
    };

    logOut = event => {
        this.props.dispatch(middlewareLogOutUser(this.props.idUser));
        event.stopPropagation();
    };

    redirectAbout = event => {
        if (this.props.location.pathname !== this.state.redirectAbout.path)
           this.setState({
               redirectAbout: {
                   ...this.state.redirectAbout,
                   redirectA: true
               }
           });
        event.stopPropagation()
    };

    redirectCabinet = event => {
        if (this.props.location.pathname !== this.state.redirectCabinet.path)
        this.setState({
            redirectCabinet: {
                ...this.state.redirectCabinet,
                redirectC: true
            }
        });
        event.stopPropagation()
    }

    add = (e) => {
        eventEmitter.emit('EventChangeWorkMode',{action: 'newProject'});
    };

    render(){

        let {redirectA} = this.state.redirectAbout;
        let {redirectC} = this.state.redirectCabinet;

        if (redirectA) return <Redirect to = {this.state.redirectAbout.path} />
        if(redirectC) return <Redirect to = {this.state.redirectCabinet.path} />
        console.log('heade');
        return (
            <header>
                <div className = 'container'>
                    <div className = 'flex-row'>
                            <div onClick = {this.redirectCabinet} className = 'header__CabinetInfo'>
                                <Icon path = {iconPath} />
                                <h3 className = 'tilteApp'>{this.state.title}</h3>
                            </div>
                            {this.props.location.pathname === '/Cabinet' ?
                                <div onClick = {this.add} className = 'header__newProject__AddButton'>
                                    <Icon path = '/img/plus.png' />
                                </div>
                                : null
                            }
                                <div  className = 'Navigator'>
                                    <Icon onClick = {this.redirectAbout} title = 'about' path = '/img/about_logo.svg' />
                                    <Icon onClick = {this.logOut} title = 'logOut' path = '/img/logOut.svg' />
                                </div>
                    </div>
               </div>
            </header>
        )
    }
}

export default connect()(withRouter(Header));