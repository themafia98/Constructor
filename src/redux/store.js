import {createStore} from 'redux';
import combineReducers from './reducers';

let broswerPluginDev = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const store = createStore(combineReducers,broswerPluginDev);
export default store;