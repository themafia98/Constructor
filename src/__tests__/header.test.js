import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import { MemoryRouter} from 'react-router-dom';
import store from '../redux/testReduxStore';
import Header from '../components/header/Header';

const HeaderComponent = renderer.create(
    <MemoryRouter>
        <Provider store = {store}>
            <Header title = 'Constructor' />
        </Provider>
    </MemoryRouter>
);

const act = renderer.act;

test("Header.js test", function(){

    let snapshotHeader = HeaderComponent.toJSON();
    expect(snapshotHeader).toMatchSnapshot();

    const logOut = HeaderComponent.root.find(button => button.props.className === 'logoutButton');
    act(()=> logOut.props.onClick({stopPropagation: () => null}));

    snapshotHeader = HeaderComponent.toJSON();
    expect(snapshotHeader).toMatchSnapshot();


        /** @redirect */
    const about = HeaderComponent.root.find(aboutButton => aboutButton.props.className === 'header__CabinetInfo');

    act(()=> about.props.onClick({stopPropagation: () => null}));

    snapshotHeader = HeaderComponent.toJSON();
    expect(snapshotHeader).toMatchSnapshot();

});