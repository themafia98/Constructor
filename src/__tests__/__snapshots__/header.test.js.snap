// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.js test 1`] = `
<header>
  <div
    className="container"
  >
    <div
      className="flex-row"
    >
      <div
        className="header__CabinetInfo"
        onClick={[Function]}
      >
        <img
          alt="icon"
          className="mainIcon"
          draggable={false}
          onClick={null}
          src="/img/logo.svg"
          title={null}
        />
        <h3
          className="tilteApp"
        >
          Constructor
        </h3>
      </div>
      <div
        className="Navigator"
      >
        <img
          alt="icon"
          className="about"
          draggable={false}
          onClick={[Function]}
          src="/img/about_logo.svg"
          title="about"
        />
        <img
          alt="icon"
          className="logoutButton"
          draggable={false}
          onClick={[Function]}
          src="/img/logOut.svg"
          title="logOut"
        />
      </div>
    </div>
  </div>
</header>
`;

exports[`Header.js test 2`] = `
<header>
  <div
    className="container"
  >
    <div
      className="flex-row"
    >
      <div
        className="header__CabinetInfo"
        onClick={[Function]}
      >
        <img
          alt="icon"
          className="mainIcon"
          draggable={false}
          onClick={null}
          src="/img/logo.svg"
          title={null}
        />
        <h3
          className="tilteApp"
        >
          Constructor
        </h3>
      </div>
      <div
        className="Navigator"
      >
        <img
          alt="icon"
          className="about"
          draggable={false}
          onClick={[Function]}
          src="/img/about_logo.svg"
          title="about"
        />
        <img
          alt="icon"
          className="logoutButton"
          draggable={false}
          onClick={[Function]}
          src="/img/logOut.svg"
          title="logOut"
        />
      </div>
    </div>
  </div>
</header>
`;

exports[`Header.js test 3`] = `null`;
